libs/ui/src/invalid_card_screen.tsx

```diff
@@ -1,18 +1,53 @@
 import React from 'react';
-import { Screen } from './screen';
+import { DippedSmartcardAuth, InsertedSmartcardAuth } from '@votingworks/types';
+
+import { fontSizeTheme } from './themes';
 import { Main } from './main';
 import { Prose } from './prose';
-import { fontSizeTheme } from './themes';
+import { Screen } from './screen';
+
+type LoggedOutReason =
+  | DippedSmartcardAuth.LoggedOut['reason']
+  | InsertedSmartcardAuth.LoggedOut['reason'];
+
+export interface Props {
+  // TODO: Update this component to support VxMark and VxScan, too
+  machine: 'VxAdmin' | 'VxCentralScan';
```

Comment:
I'm slightly wary of a pattern where we build forking logic based on machine into shared components. If the shared components know about the context in which they are used, they are no longer reusable abstractions. I don't think it's a huge deal in this case, but something to think about.

Some alternative options might be having a separate component per machine, having a prop to configure the error message, or using children to configure the error message.

Reply:
Great point - I was torn on this and tried a few different combinations of props. I initially had consumers construct messages on their own and pass those through. But I found enough redundancy across the consumers that I decided to push more logic into the component.

I think I can strike a balance by specifying defaults in the component and allowing consumers to override when needed. Will tweak

Reply:
Settled on a slightly different machine-agnostic interface! https://github.com/votingworks/vxsuite/pull/2224/commits/61344c0a94155716e8209ca8a70185114e78df14

Time will tell whether it holds up and whether we should just switch over to a simpler "give consumers full control over the message" interface